query-engine/query-engine/src/server/mod.rs

```diff
@@ -516,6 +546,14 @@ fn transaction_id(headers: &HeaderMap) -> Option<TxId> {
         .map(TxId::from)
 }
 
+fn query_timeout(headers: &HeaderMap) -> Option<Duration> {
+    headers
+        .get("X-query-timeout")
+        .and_then(|h| h.to_str().ok())
+        .and_then(|value| value.parse::<u64>().ok())
+        .map(Duration::from_millis)
+}
```

Comment:
Contextual comment:
- this signals a timeout failure to Accelerate via the `X-query-timeout` header (step 5 of the [spec](https://github.com/prisma/team-orm/issues/1212#issue-2366795687))